wallet-new/src/Cardano/Wallet/API/V1/Handlers/Accounts.hs

```diff
@@ -92,3 +94,21 @@ redeemAda :: PassiveWalletLayer IO
           -> Handler (WalletResponse Transaction)
 redeemAda _layer _wId _accIdx _redemption =
     error "unimplemented, see [CBR-349]"
+
+getAccountAddresses
+    :: PassiveWalletLayer IO
+    -> WalletId
+    -> AccountIndex
+    -> RequestParams
+    -> FilterOperations WalletAddress
+    -> Handler (WalletResponse AccountAddresses)
+getAccountAddresses _layer _wId _accIdx _pagination _filters =
+    error "unimplemented"
```

Comment:
Should we link to the ticket which is going to implement this? 

Also, genuine question -- why not reuse a `WalletResponse [WalletAddress]` , so we can do pagination the standard way?

Reply:
There's no such ticket yet :no_mouth:,about to create one.

```
why not reuse a WalletResponse [WalletAddress] , so we can do pagination the standard way?
```

Because of the way the input is returned. We made the choice of wrapping it inside a "partial" object with a single field `{ "addresses": [ ... ] }` to basically mimic the response you'd get when querying just an account. That's debatable, but that's what we've opted for, hence the wrapper.